cg/clients/chanjo2/api.py

```diff
@@ -0,0 +1,40 @@
+"""Coverage analysis API for human clinical sequencing data."""
+
+import logging
+from typing import Any
+
+import requests
+
+from cg.clients.chanjo2.models import CoverageData, CoverageRequest
+
+LOG = logging.getLogger(__name__)
+
+
+class Chanjo2APIClient:
+    """
+    Chanjo2 API to communicate with a d4tools software in order to return coverage
+    over genomic intervals.
+    """
+
+    def __init__(self, config: dict[str, str]):
+        self.headers = {
+            "Content-Type": "application/json",
+            "accept": "application/json",
+        }
+        self.host = config["chanjo2"]["host"]
+
+    def get_coverage(self, coverage_request: CoverageRequest) -> CoverageData | None:
+        """Send a POST request to the coverage endpoint to retrieve gene coverage summary data."""
+        endpoint: str = f"{self.host}/coverage/d4/genes/summary"
+        post_data: dict[str, Any] = coverage_request.model_dump()
+        try:
+            response = requests.post(url=endpoint, headers=self.headers, json=post_data)
+            response.raise_for_status()
+            response_content: dict[str, Any] = response.json().values()
+            if not response_content:
+                LOG.error("The POST get coverage response is empty")
+                return None
+            return CoverageData(**next(iter(response_content)))
```

Comment:
What is the next and iter stuff doing? Not familiar with it! If you use a Pydantic model and parse the response directly into the model as described above I think this can be skipped

Reply:
I'm pretty sure you can just do this instead
```python
json = response.json()
return CoverageData.model_validate(json)
```

Reply:
The response we get is in the format {sample_id: {coverage: ..., completeness: ...}}. I want to get rid of the sample ID, which is why I extract the first value with the iterator

Reply:
Actually the request seems odd for a specific sample as well as its return, but since the discussion in https://github.com/Clinical-Genomics/cg/pull/3267 have been going on for two months, it might be time to move forward with what we have 🥵

Reply:
I think it is better to incorporate that in the pydantic model so we have an exact map of the actual response data structure. The consumer of the client will have to discard whatever data is not needed.